json-schema/json-schema/src/main/scala/endpoints4s/Validated.scala

```diff
@@ -110,4 +112,60 @@ object Validated {
       case Failure(error) => Invalid(error.getMessage)
     }
 
+  /** Turns an `IterableOnce[Validated[A]]`into a `Validated[IterableOnce[A]]`.
+    * If all values in the sequence are `Valid`, the result will be `Valid`
+    * If there is at least one `Invalid`, the result will be an `Invalid` containing the error messages of all `Invalid`s
+    * If the `Seq` is empty, will return a `Valid[Seq.empty]`.
+    *
+    * @see [[scala.concurrent.Future.sequence]]
+    *
+    * @param in   the `IterableOnce` of Validateds which will be sequenced
+    * @tparam A   the type of the value inside the Validateds
+    * @tparam CC  the type of the `IterableOnce` of Validateds
+    * @tparam To  the type of the resulting collection
+    * @return     the `Validated` of the resulting collection
+    */
+  def sequence[A, CC[+X] <: IterableOnce[X], To](
```

Comment:
Note that CC needs to be covariant in X, otherwise the function would refuse a `Seq[Valid]` without an explicit cast.

Reply:
What is the error message? This is weird since in `traverse` the `M` type parameter is invariant and that does not seem to be an issue.

Reply:
Removing the `+` causes this error in my tests:

```
[error] .../endpoints4s/json-schema/json-schema/src/test/scala/endpoints4s/ValidatedTests.scala:155:17: no type parameters for method sequence: (in: CC[endpoints4s.Validated[A]])(implicit bf: collection.compat.BuildFrom[CC[endpoints4s.Validated[A]],A,To]): endpoints4s.Validated[To] exist so that it can be applied to arguments (Seq[endpoints4s.Valid[Product with ValidatedTests.this.FooBar with java.io.Serializable]])
[error]  --- because ---
[error] argument expression's type is not compatible with formal parameter type;
[error]  found   : Seq[endpoints4s.Valid[Product with ValidatedTests.this.FooBar with java.io.Serializable]]
[error]  required: ?CC[endpoints4s.Validated[?A]]
[error]       Validated.sequence(Seq(validFoo, validBar, validFoo)) shouldBe Valid(
[error]                 ^
[error] .../endpoints4s/json-schema/json-schema/src/test/scala/endpoints4s/ValidatedTests.scala:155:29: type mismatch;
[error]  found   : Seq[endpoints4s.Valid[Product with ValidatedTests.this.FooBar with java.io.Serializable]]
[error]  required: CC[endpoints4s.Validated[A]]
[error]       Validated.sequence(Seq(validFoo, validBar, validFoo)) shouldBe Valid(
[error]                             ^
[error] two errors found
```

It succeeds again if I make the type of Seq explicit:

```
    "turn an IterableOnce of all Valids into a Valid of an iterable" in {
      Validated.sequence(Seq[Validated[_]](validFoo, validBar, validFoo)) shouldBe Valid(
        Seq(inputFoo, inputBar, inputFoo)
      )
    }
```

Reply:
Ah, I see. I would say this is okay to have to explicitly write `Validated.sequence(Seq[Validated[FooBar]](…))` because the problem in my opinion is due to a limitation of the type inference capabilities of the Scala 2 compiler. I wonder if the type annotation would be unnecessary with Scala 3.